client/asset/interface.go

```diff
@@ -420,6 +420,14 @@ type LogFiler interface {
 	LogFilePath() string
 }
 
+// DynamicSwapFeeChecker accepts an initiation or redemption coinID and returns
+// the fee spent on the transaction along with the secrets included in the tx.
+// Returns asset.CoinNotFoundError for unmined txn. Will also error if the
+// secret in the contractData is not found in the transaction secrets.
+type DynamicSwapFeeChecker interface {
+	DynamicSwapFeesPaid(ctx context.Context, coinID, contractData dex.Bytes) (fee uint64, secretHashes [][]byte, err error)
+}
```

Comment:
Names and functionality don't really match. We typically use "swap" to signify the "initiate" or "participate" txs, not the redemption. If we want this to pull double duty, we should name it more generally, but I'd prefer separate methods for swaps and redemptions since they will be handled differently internally with regards to data parsing.

Reply:
Ok. Names are rather long but the interface implements two methods now. https://github.com/decred/dcrdex/compare/426f1421d142e92123f332020cea28bbfd8e74df..6780dde689d4493e5288e966d3b97557ab4c60e3